feat: added new /tools page

akshatnema opened the pull request:
**Description**
This PR adds a brand new page for the list of tools that give users to filter the tools according to their needs. Various filter parameters are given to filter the tools accordingly. This PR should be merged after the approval and merging of #939 so that linearly the data will be updated in the UI as well. Also, this PR will be drafted until UI of the page is not completed.


**Related issue(s)**
Resolves #383
<!-- If you refer to a particular issue, provide its number, othewise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->

akshatnema committed:
Tools UI initiated

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'asyncapi:master' into tools-dashboard

akshatnema committed:
Merge branch 'asyncapi:master' into tools-dashboard

akshatnema committed:
inital layout created

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge remote-tracking branch 'refs/remotes/origin/tools-dashboard' into tools-dashboard

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
updated tools

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
props color fixed

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
context api setup

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
Filters Context created

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'asyncapi:master' into tools-dashboard

akshatnema committed:
updated view

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
colors error fixed

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
updated view

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge remote-tracking branch 'refs/remotes/origin/tools-dashboard' into tools-dashboard

akshatnema committed:
tag colors added

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
all filters completed

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
bugs fixed

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema commented:
@derberg can you please review the PR as majority of the UI and Filters are being completed. Certain UI fixes are remaining which will be completed soon.

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
UI corrected in mobile view

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
New Blog initiated

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
blog content in progress

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
updated UI

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Blog creation in process

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
blog creation in progress

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
false commit

akshatnema committed:
Revert "false commit"

This reverts commit faa162fb4c36071719eb118e241d18e06abb2cb7.

akshatnema committed:
Merge branch 'master' into tools-dashboard

magicmatatjahu commented:
Review from UX perspective. You can fix some below bugs in another PRs, I don't have a problem with that.

---

As you can see in picture below, we have this padding on the left and right of new page which doesn't fit with content on navigation. We should remove it so left border of AsyncAPI logo should "fir" with "filter" button. I hope you understand.

![image](https://user-images.githubusercontent.com/20404945/200002814-d7ab13ac-3a8c-4965-b85b-1bb26b380643.png)

--- 

User should have possibility to hide dropdowns from `Filter` and `Select category` when will click outside of opened dropdowns - now I have to click again on button to hide dropdown, it's very bad behaviour from UX perspective

--- 

Can we add some additional margin between button and dropdown? 2x or 3x more will be good.

![image](https://user-images.githubusercontent.com/20404945/200003678-28524a35-8c46-473a-9b74-7c78f6a8683a.png)

---

Dropdown should have shadow to recognize that dropdown is separate part of UI - I had a problem, because as you can see in picture, dropdown "merges" with tool item 😅 

![image](https://user-images.githubusercontent.com/20404945/200004407-e453bf22-d03e-4e83-9bfc-9d4f5cbb84d6.png)

---

`Clear filters` button should be positioned on the right of dropdown :)

![image](https://user-images.githubusercontent.com/20404945/200004529-161d2c71-6de9-465e-9685-6e42ed4ae85a.png)

---

Probably you have to add some additional padding on the right of filter items 😄 Also I think of showing filters items in a list, one below one, not as like current, next to next. WDYT? I am opened to this proposition, and we can go with current solution  but I wonder if list will be better.

![image](https://user-images.githubusercontent.com/20404945/200004681-943b764a-4e9a-407e-937e-bdb266844ca0.png)

---

When I click on `Select` item, dropdown stretches dropdown. Items in the select should "cover" the dropdown and not stretch it - like the native select.

---

When I type something in the search engine I should see a `X` button to remove content from the search, on the right of the input. Additionally, when I don't meet search requirements I should have information something like `None tool doesn't meet search requirements` - both when I search using search input and filters. 

![image](https://user-images.githubusercontent.com/20404945/200005822-ad64945f-1014-47ca-9c30-963fbfd4f340.png)

---

I don't know why, but when I open dropdowns the icons load dynamically and stretch the content in the dropdowns - very bad UX feel.

---

`Select category` should not be greyed when I hover dropdown. Also we should rename label from `Select category` to `Select categories`.

![image](https://user-images.githubusercontent.com/20404945/200006884-6501b9c7-0d51-4ac5-886b-4fadde24162a.png)

---

- `Show more` doesn't work 😅 Additionally, you can check on `asyncapi-asciidoc-template` tool that it has left, not-eyes friendly, padding on the `Show more`.
- `Free` label should be placed always on the right.
- Labels for `Languages` and `Technology` should be clickable and we should enable filters for them (when user click them) and maybe also change colors for "active" labels? WDYT?
- Can we make those buttons for github and external site smaller? I know it's like that on the design but they are too big compared to the content and distract from the rest of the information.

![image](https://user-images.githubusercontent.com/20404945/200007183-1a1d8d35-748f-41df-a9d9-6fa6ae57987f.png)

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
changes made

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
close button added

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
changes made according to review

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema commented:
> `Clear filters` button should be positioned on the right of dropdown :)

Do you want to align this with each dropdown like `Languages` and `Technologies`? I made this button to clear all the filters that modal contains (except category filter).  I think that's what it is there for in design, to clear all filters in webpage.

> When I type something in the search engine I should see a X button to remove content from the search, on the right of the input. Additionally, when I don't meet search requirements I should have information something like None tool doesn't meet search requirements - both when I search using search input and filters.

I added `X` button and condition when none of the tools match the filters, but I couldn't able to get good UI for this situation. Right now, you can see the text coming up. Can you suggest any?

> I don't know why, but when I open dropdowns the icons load dynamically and stretch the content in the dropdowns - very bad UX feel.

I couldn't get what you want to point out here. How the icons load dynamically, they are already saved inside the repo.

> When I click on Select item, dropdown stretches dropdown. Items in the select should "cover" the dropdown and not stretch it - like the native select.

What I felt in this one is that if we consider native select design, we can come into a problem it can have a long vertical scroll inside the Select input. Currently, we can improve this a little bit more to have a horizontal scroll feature, if the number of options increases. Secondly, I only worked around the design which is made and it looks good to me.

> Labels for Languages and Technology should be clickable and we should enable filters for them (when user click them) and maybe also change colors for "active" labels? WDYT?

Probably, this can be added, but I have taken this feature to be as a part of future PR because we are still in the part of initial development of the Dashboard and we should not left any bug for at least first release of this. Later on, probably we can add more features and improve the UI of the dashboard, as per the users' feedback.

> Can we make those buttons for github and external site smaller? I know it's like that on the design but they are too big compared to the content and distract from the rest of the information.

I don't found any other way to make them smaller, because it should be similar to the UI I made, so if you have any design in your mind or anyone can suggest, I can improve them.


@magicmatatjahu for rest of them, I've done with my changes. You can review them again. For now, I'm making this PR out for ready for review, but it still contains an incomplete blog that I'll complete in a few days. @derberg Looking for a review from your side :+1:.

akshatnema committed:
Blog updated

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

derberg committed:
Merge branch 'master' into tools-dashboard

derberg commented:
imho blog post should follow in a separate PR. It adds to much noise here and also it has different focus.

this PR should also migrate https://deploy-preview-940--asyncapi-website.netlify.app/docs/tools to the new tools. UI is beautiful but hard to do any manual testing as there are just 2 tools

derberg commented:
also looking at:

```
Here you can quickly discover various tools to optimize your journey with AsyncAPI! These tools are made by the community, for the community. Have an AsyncAPI tool that you want featured on this list? [Submit it here!](https://github.com/asyncapi/website/issues/new?template=blog.md)
```

the link points to the issue creation template. Instead, it should be a markdown document that describes how things work

akshatnema commented:
> imho blog post should follow in a separate PR. It adds to much noise here and also it has different focus.

Yeah, perfectly understand having a separate PR but if we want to merge the changes of the Tools Dashboard, we should also explain to the community what are we focusing on with this and how to contribute in adding Tools to the website.

> this PR should also migrate [deploy-preview-940--asyncapi-website.netlify.app/docs/tools](https://deploy-preview-940--asyncapi-website.netlify.app/docs/tools) to the new tools. UI is beautiful but hard to do any manual testing as there are just 2 tools

Yeah, if there is an insertion of a new tool in GitHub, it will update itself in a week because the respective GitHub action works on Monday of each week.

> Instead, it should be a markdown document that describes how things work

But how we gonna link markdown document to it? Are you talking about the documentation?

derberg commented:
> Yeah, perfectly understand having a separate PR but if we want to merge the changes of the Tools Dashboard, we should also explain to the community what are we focusing on with this and how to contribute in adding Tools to the website.

but nobody will see it as we will not enable it in navigation. We will enable it with your blog post PR.
Advantage is that if it is silently released first, we can also test it with some repos from AsyncAPI org, and also ask few external folks to use new solution.

> But how we gonna link markdown document to it? Are you talking about the documentation?

yes, markdown document in community repo

akshatnema commented:
> but nobody will see it as we will not enable it in navigation. We will enable it with your blog post PR.

I can add it in the navigation right now. I think I did majority of the UI implementation and it is ready to test it out with AsyncAPI org repos.

> Advantage is that if it is silently released first, we can also test it with some repos from AsyncAPI org, and also ask few external folks to use new solution.

We can still test it in this PR only. Just make new files in the respective Tools repo and github action is already working on master branch. The changes will reflect on this PR in a week. Even if you want the changes immediately, ping me out when you are done adding the `.asyncapi-tool` file and I'll manually run the scripts in this PR.

> yes, markdown document in community repo

So, the official documentation of this addition of tools will exist in the form of md file? I thought of making this documentation available on website so to provide better reach and availability to all the users.

akshatnema committed:
Merge branch 'master' into tools-dashboard

derberg commented:
Imho better to test on production with functionality hidden from navigation. As it was tested with scripts. And fixes have follow-up PRs. But if you prefer long-hanging PR, up to you. But brace yourself for merge conflicts 😄 

and yes, it should go to community, as it is related to community, and later, once we have the new `community` section in website, we will display these document there in the UI

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema commented:
> Imho better to test on production with functionality hidden from navigation. As it was tested with scripts. And fixes have follow-up PRs. But if you prefer long-hanging PR, up to you. But brace yourself for merge conflicts.

There isn't any issue with merging conflicts for me :sweat_smile:, but yeah, if you want to test it out in production, then ok, we can merge this PR first and then, can have subsequent PRs to handle bugs. I'll save the article in some other folder :smile:. But also, I would like to come up with the suggestion of @magicmatatjahu who asked to have a separate branch of `tools` in the repo where we can merge this PR and then have branch deploys to see the changes according to production. But, with this idea, we still have to merge the conflicts between the branches. What is your take on this?

akshatnema committed:
removed blog

akshatnema committed:
Merge remote-tracking branch 'refs/remotes/origin/tools-dashboard' into tools-dashboard

akshatnema commented:
@derberg I have removed the blog from the PR. You can now go on to merge the PR.

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
illustration added for no tools found

quetzalliwrites committed:
Merge branch 'master' into tools-dashboard

quetzalliwrites commented on pages/tools/index.js:
```diff
@@ -0,0 +1,38 @@
+              Here you can quickly discover various tools to optimize your
+              journey with AsyncAPI! These tools are made by the community, for
+              the community. Have an AsyncAPI tool that you want featured on
+              this list?
```
```suggestion
              Discover various AsyncAPI tools to optimize your journey! 
              These tools are made by the community, for the community.
              Have an AsyncAPI tool you want to be featured on this list?
```

quetzalliwrites commented on pages/tools/index.js:
```diff
@@ -0,0 +1,38 @@
+              the community. Have an AsyncAPI tool that you want featured on
+              this list?
+              <TextLink href="https://github.com/asyncapi/website/issues/new?template=blog.md" target="_blank">
+                Submit it here!
```
```suggestion
                Submit new AsyncAPI tools here!
```

For **accessibility**, you should never write anchor text that doesn't tell the user where a link will take them. (Source: [WACAG](https://www.w3.org/TR/WCAG20-TECHS/H30.html))

IOW, you should not write anchor text such as:
- submit here
- read more
- click here
- this
- etc.

quetzalliwrites commented:
This looks great! Nice work so far.

I see an issue to address. We cannot assume that users will find this new `/Tools` dashboard unless we enable click-through interaction on the menu item for `<li>Tools</li>`.  (Adding a new sub-menu item under the **Tools** menu isn't a great DevEx because the dashboard is not a tool, it's more of a homepage.)

<img width="400" alt="Screen Shot 2022-11-11 at 5 56 43 PM" src="https://user-images.githubusercontent.com/19964402/201451169-57e6b271-9175-43ac-b5d0-c10a19561067.png">

quetzalliwrites reviewed (changes requested):
review left 👍🏽

akshatnema commented:
> I see an issue to address. We cannot assume that users will find this new /Tools dashboard unless we enable click-through interaction on the menu item for `<li>Tools</li>`.

@alequetzalli  We aren't allowing this dashboard for public right now as according to @derberg, we still need testing on this PR, so in the final PR (basically, it is of blog), we will be discussing where the link to the Dashboard should be added.

akshatnema committed:
Apply suggestions from code review

Co-authored-by: Alejandra Quetzalli  <alejandra.quetzalli@postman.com>

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema commented:
@magicmatatjahu PR is up for the next review. Kindly review this PR on the basis of code changes I did (as you mentioned earlier that you didn't in the previous review).

quetzalliwrites committed:
Merge branch 'master' into tools-dashboard

quetzalliwrites commented:
> > I see an issue to address. We cannot assume that users will find this new /Tools dashboard unless we enable click-through interaction on the menu item for `<li>Tools</li>`.
> 
> @alequetzalli We aren't allowing this dashboard for public right now as according to @derberg, we still need testing on this PR, so in the final PR (basically, it is of blog), we will be discussing where the link to the Dashboard should be added.

aaah got it, makes sense.... looking good, nice job so far!

derberg commented:
The only fix for this PR is related to `View on GitHub` button. We allow projects from other sources, like `asyncapi-asciidoc-template` that you have there in sample data. So minimum we need is `View on GitHub` for projects located on `github` and in case of projects that are not located on github, we need alternative `View Source Code` without github icon.

---

@magicmatatjahu you agree going with it to production the way it is hidden from navigation?

@akshatnema so just to make sure, next steps you have on list is:
- documentation in community repo
- migration of https://deploy-preview-940--asyncapi-website.netlify.app/docs/tools to manual json?
- blog post

What else?

And we can already start adding tools file to repos, as the current structure is fine and won't change for now?

akshatnema commented:
> So minimum we need is View on GitHub for projects located on github and in case of projects that are not located on github, we need alternative View Source Code without github icon.

Fine 👍, so I have to check using regex on the frontend side, whether the given code link is of GitHub or not, right? or we should have another parameter in the JSON object of a tool to check whether it is on GitHub or not?

> migration of [deploy-preview-940--asyncapi-website.netlify.app/docs/tools](https://deploy-preview-940--asyncapi-website.netlify.app/docs/tools) to manual json?

Now, the question arises that are we going to fully migrate the [deploy-preview-940--asyncapi-website.netlify.app/docs/tools](https://deploy-preview-940--asyncapi-website.netlify.app/docs/tools) to `manual.json`? Because there are certain tools specified which can easily be added using the `.asyncapi-tool` file. So, I would suggest to first add most of the tools by `.asyncapi-tool` file and then we can focus on adding the tools via `manual.json` file accordingly using PRs. What's your opinion in this? @derberg 

> blog post

Blog Post is almost ready and it only require certain fixes in the content. I had added that post in this PR first but as you suggested that it should be in next PR, I have removed it from this.

> documentation in community repo

I'll add the documentation probably in the next month, after my exams get over.

> And we can already start adding tools file to repos, as the current structure is fine and won't change for now?

Yes you can start adding according to current structure

akshatnema committed:
Merge branch 'master' into tools-dashboard

derberg commented:
> Fine 👍, so I have to check using regex on the frontend side, whether the given code link is of GitHub or not, right? or we should have another parameter in the JSON object of a tool to check whether it is on GitHub or not?

regex/split, whatever makes code less complicated. No need for some explicit.
I personally would do:
```js
const url = new URL('https://github.com/akshatnema/Login-Registration-project')
// this gives you access to url.host which returns github.com
const hostName = url.host;
const isGitHub = hostName === 'github.com'
// now you can easily make a condition how to display info about source code
```

> Now, the question arises that are we going to fully migrate the [deploy-preview-940--asyncapi-website.netlify.app/docs/tools](https://deploy-preview-940--asyncapi-website.netlify.app/docs/tools) to manual.json? Because there are certain tools specified which can easily be added using the .asyncapi-tool file. So, I would suggest to first add most of the tools by .asyncapi-tool file and then we can focus on adding the tools via manual.json file accordingly using PRs. What's your opinion in this? @derberg

the ones that are not in `asyncapi` org yes, we need to migrate them. Then it is up to owners if they wanna switch.

> Yes you can start adding according to current structure

perfect, I'll start with the ones that are in current list of tools, to feel how it goes

derberg commented:
Looks like automation workflow do not work as expected -> https://github.com/asyncapi/website/actions/runs/3510469460/jobs/5880294792

It turns green but in `Regenerate` script, when some tool file is broken, the error message is `Error: [object Object]`. Also pull request step do not work


EDIT: easier and faster it was to open a PR than complain 😄 https://github.com/asyncapi/website/pull/1128

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema commented:
> It turns green but in Regenerate script, when some tool file is broken, the error message is Error: [object Object]. Also pull request step do not work

@derberg Yeah, this workflow won't make a red signal or will not make errors in between execution of the workflow, because if someone else made the `.asyncapi-tool` file wrong, the workflow should make the file with the remaining tools information extracted from the API. And  the error messages we get should be notified to us via Slack, but not stop the execution of file. Pull Request is not made because the same file already exists in the master branch, no need to create the PR.

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
View Source code button added

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema commented:
@magicmatatjahu I need a review from your side regarding the implementation/code changed in the PR. Probably, we have to finish this review in this week only so that we can test the tools in the week by merging this with the master branch.

akshatnema committed:
Merge branch 'master' of github.com:akshatnema/website

akshatnema committed:
Merge branch 'master' into tools-dashboard

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'master' into tools-dashboard

derberg commented:
@akshatnema can you solve conflicts and update branch. I'd like to do manual testing with latest additions of `.asyncapi-tool`

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema commented:
@derberg have you added any new tools till now or just testing on previous ones?

derberg commented:
@akshatnema no new once, just those that are in `master` already. Maybe this is enough to manually test UI

@magicmatatjahu you can add for studio and server-api 😉

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
tools file updated

akshatnema committed:
repoUrl added in schema

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Dashboard updated

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
updated files

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
removed token

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
updated files

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

magicmatatjahu commented:
When I select for example `java` filter and click `apply filters` the website is broken and I see this 

![image](https://user-images.githubusercontent.com/20404945/207377027-31ca1681-93cd-4d2c-a44f-02b1b9177d71.png)

---

I think that dropdown content should also have border on left, right and bottom (as select has)

![image](https://user-images.githubusercontent.com/20404945/207377396-b68c42b2-b263-4cbd-bd79-469bcaf3d52a.png)

---

When I click `Show more` I see this

![image](https://user-images.githubusercontent.com/20404945/207377784-ea24695f-5177-4c43-97a6-cdee89544e7a.png)

I thought that `Show more` will expand more content, not display it in "tooltip"

--- 

When I click on `Submit new AsyncAPI...` I go to the https://github.com/asyncapi/website/issues/new?template=blog.md link - we should create separate issue template for tool suggestion.

---

I already wrote this once (as I remember), but I don't see the point of having categories as a separate dropdown, as categories, languages and technologies are filters and should be together, in one dropdown, not two. But it can stay if it only bothers me.

magicmatatjahu commented on components/icons/ArrowDown.js:
```diff
@@ -0,0 +1,7 @@
+export default function ArrowDown({ className= '' }) {
```
```suggestion
export default function IconArrowDown({ className= '' }) {
```

magicmatatjahu commented on components/icons/Filter.js:
```diff
@@ -0,0 +1,14 @@
+export default function Filter({className=''}) {
```
```suggestion
export default function IconFilter({className=''}) {
```

magicmatatjahu commented on components/icons/Search.js:
```diff
@@ -0,0 +1,7 @@
+export default function SearchIcon({className = ''}) {
```
```suggestion
export default function IconSearch({className = ''}) {
```

We use the `Icon{...}` template for the icon names.

magicmatatjahu commented on components/tools/Filters.js:
```diff
@@ -0,0 +1,107 @@
+      filter: false,
+      category: false
+    })
+  }
```
I guess that `Clear filters` is only visible in first dropdown, so if I check some categories (in the second dropdown) and I wanna clear them I need to open first, yes? Very strange, we should add `Clear filters` button in the second dropdown :)

akshatnema replied:
So, according to you, the `Category` filter should come inside `Filters` menu? If we make it this way, the `Clear Filter` button will function to remove categories from filters too, but then what should be replaced by that menu in the bar?

magicmatatjahu replied:
> but then what should be replaced by that menu in the bar?

Nothing, it should be removed.

magicmatatjahu commented on components/tools/Filters.js:
```diff
@@ -0,0 +1,107 @@
+    <div className="bg-white z-10 py-4 border rounded-lg border-gray-300 shadow-md">
+      <div className="flex flex-col gap-2 mx-4">
+        <div className="flex gap-2 items-baseline justify-between">
+          <div className="text-sm text-gray-500">PRICING</div>
```
You can use `uppercase` class:

```suggestion
          <div className="text-sm text-gray-500 uppercase">Pricing</div>
```

magicmatatjahu commented on components/tools/FiltersDropdown.js:
```diff
@@ -0,0 +1,28 @@
+      tempValueArray.push(data);
+    }
+    setStateFunction(tempValueArray)
+    e.stopPropagation();
```
`e.stopPropagation();` should be fired at the beginning of function. Of course it's not a "wrong", rather a "bad" decision.

magicmatatjahu commented on components/tools/FiltersDropdown.js:
```diff
@@ -0,0 +1,28 @@
+
+export default function FiltersDropdown({dataList=[], checkedOptions=[], setStateFunction, className=''}) {
+
+  const handleClickOption = (e, data) => {
```
```suggestion
  const handleClickOption = (e, option) => {
```

:)

magicmatatjahu commented on components/tools/ToolDashboard.js:
```diff
@@ -0,0 +1,155 @@
+    const [searchName, setSearchName] = useState('')
+    const [toolsList, setToolsList] = useState({})
+    const [checkedCategory, setCheckedCategory] = useState(categories)
+    const [checkToolsList, setcheckToolsList] = useState(true)
```
```suggestion
    const [checkToolsList, setCheckToolsList] = useState(true)
```

magicmatatjahu commented on components/tools/ToolDashboard.js:
```diff
@@ -0,0 +1,155 @@
+export default function ToolDashboard() {
+    const filterRef = useRef()
+    const categoryRef = useRef()
+    const [openFilter, setopenFilter] = useState({
```
```suggestion
    const [openFilter, setOpenFilter] = useState({
```

magicmatatjahu commented on components/tools/ToolDashboard.js:
```diff
@@ -0,0 +1,155 @@
+            })
+            if(tempToolsList[category].toolsList.length) setcheckToolsList(true)
+        })
+        setToolsList(tempToolsList)
```
Probably you should put that logic to separate function 😅

akshatnema replied:
Done 👍

magicmatatjahu commented on components/tools/ToolDashboard.js:
```diff
@@ -0,0 +1,155 @@
+        let newFilterObject = { ...openFilter };
+        if (filterType === 'filter') {
+            newFilterObject.filter = !newFilterObject.filter
+            newFilterObject.category = false
```
```suggestion
            newFilterObject = {
              filter: !newFilterObject.filter,
              category: false,
            };
```

magicmatatjahu commented on components/tools/ToolsCard.js:
```diff
@@ -0,0 +1,107 @@
+import Tag from './Tags';
+import TextTruncate from 'react-text-truncate';
+
+export default function toolsCard({ toolData }) {
```
```suggestion
export default function ToolsCard({ toolData }) {
```

magicmatatjahu commented on components/tools/ToolsCard.js:
```diff
@@ -0,0 +1,107 @@
+
+export default function toolsCard({ toolData }) {
+  const [showDescription, setShowDescription] = useState(false)
+  const [onGithub, setonGithub] = useState(false)
```
```suggestion
  const [onGithub, setOnGithub] = useState(false)
```

magicmatatjahu commented on components/tools/ToolsCard.js:
```diff
@@ -0,0 +1,107 @@
+  useEffect(() => {
+    const url = new URL(toolData.links.repoUrl)
+    if (url.host == 'github.com') setonGithub(true)
+    else setonGithub(false)
```
You don't need to make it as useEffect, you can perform logic in separate function, or even outside that useEffect - and remove `const [onGithub, setOnGithub] = useState(false)` useState.

magicmatatjahu commented on pages/tools/index.js:
```diff
@@ -0,0 +1,37 @@
+
+export default function toolsDashboard() {
+  const description = 'Tools Dashboard for AsyncAPI Initiative';
+  const image = '/img/social/meetings.png';
```
We need new image for new page, please remember about that.

akshatnema replied:
Means? Should I copy this image file and rename it as `tools.png`?

magicmatatjahu replied:
No, you should create new image. Check how `meetings.png` looks like - it's related to the meetings, not tools.

akshatnema replied:
But almost every page has the same AsyncAPI logo shown in the tab as the favicon, then why create a new one?

akshatnema committed:
paid tools corrected

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
Apply suggestions from code review

Co-authored-by: Maciej Urbańczyk <urbanczyk.maciej.95@gmail.com>

akshatnema committed:
changes according to review

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema commented:
@magicmatatjahu Thanks for reviewing the PR. I have made the changes according to your review but there are certain comments we have to discuss.

> When I click on Submit new AsyncAPI... I go to the [asyncapi/website/issues/new?template=blog.md](https://github.com/asyncapi/website/issues/new?template=blog.md) link - we should create separate issue template for tool suggestion.

We can't provide any issue template for tools as for now because if anyone wants to add a tool even manually, he has to make PR regarding the addition of a new JSON object inside `tools-manual.json`. Hence, I didn't find anything useful to add to this link, so it's better we remove this line. WDYT? @derberg 

> I thought that Show more will expand more content, not display it in "tooltip"

Sorry, but we can't expand the size of the tool because it will change the size of the tool cards aligned horizontally with it, which will make the website behave awkwardly. If a certain contributor adds a long description of a tool, we can't expand the size of tool so much to cover it inside that card only. It's better to expand the description inside a separate modal.

Some comments are left in the review above so do check them out. Rest, the changes have been made, kindly look at them for further review.

magicmatatjahu commented:
@akshatnema 

> We can't provide any issue template for tools as for now because if anyone wants to add a tool even manually, he has to make PR regarding the addition of a new JSON object inside tools-manual.json. Hence, I didn't find anything useful to add to this link, so it's better we remove this line. WDYT? @derberg

So we should remove it or point in the link to the instruction how to add tools.

> Sorry, but we can't expand the size of the tool because it will change the size of the tool cards aligned horizontally with it, which will make the website behave awkwardly. If a certain contributor adds a long description of a tool, we can't expand the size of tool so much to cover it inside that card only. It's better to expand the description inside a separate modal.

So we should display the entire description and not how it currently looks. If you do not want to display the entire description, there are two options (better than current):

- a tooltip which shows the description when hovered over, not when clicked on, and which disappears when someone moves the mouse away from the `Hover for more information` button.
- a separate modal (as you suggest) but for the whole page and with all the details of the tool.

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
Merge branch 'master' into tools-dashboard

derberg commented:
> So we should remove it or point in the link to the instruction how to add tools.

yup modify in the way that it makes sense that the link you put is the link to instruction you provide for the community repo

> Sorry, but we can't expand the size of the tool because it will change the size of the tool cards aligned horizontally with it, which will make the website behave awkwardly. If a certain contributor adds a long description of a tool, we can't expand the size of tool so much to cover it inside that card only. It's better to expand the description inside a separate modal.

I think we need to be consistent across pages. Look at https://deploy-preview-940--asyncapi-website.netlify.app/community/tsc. If box have more content and need to get bigger, the whole row gets bigger, and we show all, no `show more`

akshatnema commented:
> I think we need to be consistent across pages. Look at [deploy-preview-940--asyncapi-website.netlify.app/community/tsc](https://deploy-preview-940--asyncapi-website.netlify.app/community/tsc). If box have more content and need to get bigger, the whole row gets bigger, and we show all, no show more

But what if a certain person makes a very long description by mistake, in this case, we should not make our UI awkward or uneven. Showing a description of around 50 words is not a good option for a particular-sized Tool Card. I will prefer the idea of @magicmatatjahu to have a new Modal for expanding a Tool in Dashboard, but we don't have a design ready for that, and at the same time, no detailed information about the tool to show up the tool in the new Model. So, we can have hover type `tooltip` to show the description in the same way as of now, instead of on click. What do you prefer to have?

akshatnema committed:
replaced issue link

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
category filter moved to Filter

derberg commented:
@akshatnema true, this is indeed possible, you are right. Just make sure that we show by default at lease 3-5 lines and then `show more`

derberg commented on scripts/tools/tags-color.js:
```diff
@@ -1,37 +1,172 @@
+        borderColor: "border-[#EC2125]"
     },
     {
         name: "Javascript",
```
```suggestion
        name: "JavaScript",
```

derberg commented on scripts/tools/tags-color.js:
```diff
@@ -1,37 +1,172 @@
+        borderColor: "border-[#3878AB]"
+    },
+    {
+        name: "Typescript",
```
```suggestion
        name: "TypeScript",
```

akshatnema committed:
Apply suggestions from code review

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

akshatnema committed:
dynamic addition of technologies and languages in the filters

akshatnema committed:
Merge branches 'tools-dashboard' and 'tools-dashboard' of github.com:akshatnema/website into tools-dashboard

akshatnema committed:
added tooltip

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema committed:
comments added

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema commented:
@mcturco Kindly review this PR from a design perspective as we also require clarity from a UI perspective that the implementation of the tools page is correct.

Preview link - https://deploy-preview-940--asyncapi-website.netlify.app/tools

akshatnema committed:
comments added for clarity

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
updated the tools.json file

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

quetzalliwrites commented:
I know I'm no designer, but I wanted to leave a comment to compliment this page's design. I find it lovely and with a good UX. ✨

derberg commented:
I second what @alequetzalli wrote. The difference is that #designismypassion and you know that once I say UI is great, it is great indeed 🤣 🤣 🤣 🤣 🤣 

I'd only like to suggest one change that always comes to my mind when I see `Free`. My suggestion:
- rename `Free` to `Open-source` or `FOSS`
- rename `Paid` to `Commercial`

wdyt?

akshatnema commented:
> wdyt?

From my perspective, everything is ok because I just need to change a single line of text 😂. But definitely, we should ask @magicmatatjahu @alequetzalli regarding what's their view on these terms. They can give us more clarity over the terms we should use here.

akshatnema committed:
Merge branch 'master' into tools-dashboard

magicmatatjahu commented:
@akshatnema Please add separate between last select and button:

![image](https://user-images.githubusercontent.com/20404945/208942709-7e9970be-6bfc-44c5-aceb-d9394699d03f.png)

When I click on `Filter` button and then click outside, the dropdown should hide, but atm is still visible.

magicmatatjahu commented:
And last thing, border radius has bad colors 😄 
![image](https://user-images.githubusercontent.com/20404945/208943190-3037a7ce-4224-48ea-8bc1-c10f35b021f3.png)

magicmatatjahu commented:
I'm in favour of `Open-source` and `Commercial`

akshatnema commented:
@magicmatatjahu @derberg A last query 😅, where should we include Tools Dashboard option in the Navbar?

magicmatatjahu commented:
Change link for the `Tools & Services`

![image](https://user-images.githubusercontent.com/20404945/208945050-48603d1e-7967-46bc-91ec-e0e520935dfd.png)

akshatnema committed:
changes applied

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

akshatnema commented:
@derberg @magicmatatjahu all changes done, ready for review...

akshatnema committed:
changes in categories

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

quetzalliwrites committed:
Merge branch 'master' into tools-dashboard

quetzalliwrites commented:
> From my perspective, everything is ok because I just need to change a single line of text 😂. But definitely, we should ask @magicmatatjahu @alequetzalli regarding what's their view on these terms. They can give us more clarity over the terms we should use here.

IN A SHOCKING MOVE, ALEJANDRA AGREED WITH @derberg 😂

ya... **Open Source** and **Commercial** is more accurate ✌🏽

quetzalliwrites reviewed (dismissed):
SOOOO PURRRTY 😍😍😍

akshatnema committed:
Merge branch 'master' into tools-dashboard

akshatnema committed:
removed dashboard from Nav

Signed-off-by: akshatnema <20bcs022@iiitdmj.ac.in>

derberg reviewed (approved):
Yay!!! merging

@magicmatatjahu @alequetzalli the only change after your last approval was that I asked @akshatnema to "hide" this new view for now, so for now navigation still points to `/docs/tools`.

Reason is that for now new tools view is pretty empty. In another PR @akshatnema will try to migrate table from `/docs/tools` to `/tools` and then we enable the view and set redirects (if needed - probably not) for old view

derberg commented:
/rtm

quetzalliwrites commented:
> Yay!!! merging
> 
> @magicmatatjahu @alequetzalli the only change after your last approval was that I asked @akshatnema to "hide" this new view for now, so for now navigation still points to `/docs/tools`.
> 
> Reason is that for now new tools view is pretty empty. In another PR @akshatnema will try to migrate table from `/docs/tools` to `/tools` and then we enable the view and set redirects (if needed - probably not) for old view

ooooooh I see... cool! that makes sense

thanks for the extra context because I love learning and understanding changes happening
